### PR TITLE
Moving more Tfm specific properties to the targetFramework.props file

### PR DESCRIPTION
**Pull request by Anipik**

### CODE REVIEW

**Review by ericstj — APPROVED** [review]
This is a good step in the right direction.  Did you look at every property that was moved to ensure there weren't further derived properties?  Assuming so, then a next step could be to try to eliminate all usage of the properties now in `targetframework.props` from propertygroups in props files / project bodies (so that we can support defining TargetFramework in the project body).

**Review by ViktorHofer — APPROVED** [review]
Thanks for doing this. Can you please file an issue for the follow-up work which is making sure that `TargetFramework` is never read before the project file is loaded?

**Comment by Anipik** [comment]
> did you look at every property that was moved to ensure there weren't further derived properties?

The only tfm properties are left for manual shims. We need to keep them to add manual keyword in the path to avoid having the conflicts with generated shims

> Can you please file an issue for the follow-up work which is making sure that TargetFramework is never read before the project file is loaded?

yeah i will do that.

File: src/libraries/restore/Directory.Build.props [hunk 1]
```diff
@@ -7,11 +7,8 @@
     <ProjectAssetsFile>$(RestoreOutputPath)/project.assets.json</ProjectAssetsFile>
     <MSBuildProjectExtensionsPath>$(IntermediateOutputPath)</MSBuildProjectExtensionsPath>
     <SkipDeriveTargetFrameworks>true</SkipDeriveTargetFrameworks>
-    <!-- let us control the output path -->
-    <OutputPath>$(BaseOutputPath)$(TargetFramework)-$(TargetFrameworkSuffix)-$(Configuration)</OutputPath>
-    <OutputPath Condition="'$(TargetFrameworkSuffix)' == ''">$(BaseOutputPath)$(TargetFramework)-$(Configuration)</OutputPath>
     <AppendRuntimeIdentifierToOutputPath>false</AppendRuntimeIdentifierToOutputPath>
-    <AdditionalBuildTargetFrameworks Condition="'$(BuildAllProjects)' == 'true'">netstandard2.0-$(TargetFrameworkSuffix)</AdditionalBuildTargetFrameworks>
+    <AdditionalBuildTargetFrameworks Condition="'$(BuildAllProjects)' == 'true'">netstandard2.0</AdditionalBuildTargetFrameworks>
```
Comment (ViktorHofer) [inline src/libraries/restore/Directory.Build.props:11]: Why are you removing the TargetFrameworkSuffix here?

Comment (ViktorHofer) [reply src/libraries/restore/Directory.Build.props:11]: What about `RestoreOutputPath`, `ProjectAssetsFile` and `MSBuildProjectExtensionsPath` above, which all derive from `IntermediateOutputPath` which derives from `TargetFramework`.

Comment (Anipik) [reply src/libraries/restore/Directory.Build.props:11]: As eric mentioned , the next step is to remove the usage of all properties in targetFramework.props from propertyGroups.
eg all these properties will be the candidate for that.

Comment (Anipik) [reply src/libraries/restore/Directory.Build.props:11]: > Why are you removing the TargetFrameworkSuffix here?

This was a bug. TargetFrameworkSuffix is a property for innerBuilds and is not defined for outer builds and should not be used in outer build. AdditionalBuildTargetFrameworks is used only in outerbuilds.

Comment (ViktorHofer) [reply src/libraries/restore/Directory.Build.props:11]: I'm unsure what's the scope of this PR. Isn't the goal to fix the broken intermediate/output paths? An example:

image

Comment (Anipik) [reply src/libraries/restore/Directory.Build.props:11]: The scope is just cleanup, nothing is getting fixed here.

**Pull request merged**